lib/types/optional_caps_list.ex

```diff
@@ -15,6 +15,6 @@ defmodule Kadena.Types.OptionalCapsList do
   @impl true
   def new(clist \\ nil)
   def new(nil), do: %__MODULE__{}
-  def new(%CapsList{} = clist), do: %__MODULE__{clist: clist}
+  def new([%Cap{} | _tail] = clist), do: %__MODULE__{clist: clist}
```

Comment:
is the `_tail` required? I mean, for this specific case, maybe it's better to use a guard? more descriptive?
```suggestion
  def new(clist) when is_list(clist), do: %__MODULE__{clist: clist}
```
What do you think? @miguelnietoa 🤔🤔

Reply:
IMO it is better to keep `[%Cap{} | _tail]` as it validates that a list type containing Cap structs is provided to the `new/1` function

Reply:
yeah makes sense, my mistake 🙏🏻